### DISCUSSION TITLE
How to specify kubernetes version?

### DISCUSSION THREAD

**Discussion by Spaghetto392** [siderolabs/talos#8982] [Q&A] [answered]
Hi, I'm trying to setup a cluster on Talos 1.7, however, a deployment I want to run needs Kubernetes to be running on version 1.28 or lower. How do I specify which version of Kubernetes I have installed on Talos?

**Comment by steverfrancis** [comment, marked as answer]
You can define this when generating your machine configs.
see https://www.talos.dev/v1.7/introduction/prodnotes/#customizing-machine-configuration
for example
talosctl gen config --kubernetes-version 1.28.2

**Reply by Spaghetto392** [reply]
Thank you for your answer. I'm surprised I didn't see that while looking through the docs.
